src/Grid/grid.jl

```diff
@@ -665,27 +667,40 @@ end
 
 """
     getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::Int)
-Fills the vector `x` with the coordinates of a cell, defined by its cell id.
+    getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::AbstractCell)
+
+Fills the vector `x` with the coordinates of a cell defined by either its cellid or the cell object itself.
 """
-@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::Int) where {dim,T}
-    #@assert length(x) == N
+
+@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::Ferrite.AbstractGrid, cellid::Int) where {dim,T} 
+    cell = getcells(grid, cellid)
+    getcoordinates!(x, grid, cell)
+end
+
+@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::Ferrite.AbstractGrid, cell::Ferrite.AbstractCell) where {dim,T}
     @inbounds for i in 1:length(x)
-        x[i] = grid.nodes[grid.cells[cell].nodes[i]].x
+        x[i] = getcoordinates(getnodes(grid, cell.nodes[i]))
```

Comment:
This is not for this PR, but in general a `Cell` Interface function to return the nodes would be nice, like `getnodes(cell::AbstractCell,i::Integer)` I think there is none currently

Reply:
There are indeed a few things which can be improved regarding the element and mesh interface, but I would leave them to separate PRs.